AnyMap cannot be compiled with system_yamlcpp on Ubuntu 18.04

**Problem description**

Using the system `libyaml-cpp-dev` on Ubuntu 18.04 results in a compilation error for `AnyMap.cpp`. The version of `libyaml-cpp-dev` appears to be 0.5.2: https://packages.ubuntu.com/bionic/libyaml-cpp-dev

**Steps to reproduce**

1. `apt install libyaml-cpp-dev`
2. `scons build system_yamlcpp=y`
3. 
```
g++ -o build/src/base/AnyMap.os -c -std=c++0x -pthread -O3 -Wno-inline -g -Wall -include src/pch/system.h -fPIC -DNDEBUG -Iinclude -Iinclude/cantera/ext -Ibuild/src -I/usr/include/eigen3 src/base/AnyMap.cpp
src/base/AnyMap.cpp: In static member function 'static bool YAML::convert<Cantera::AnyMap>::decode(const YAML::Node&, Cantera::AnyMap&)':
src/base/AnyMap.cpp:224:28: error: 'const class YAML::Node' has no member named 'Mark'
         target.setLoc(node.Mark().line, node.Mark().column);
                            ^~~~
src/base/AnyMap.cpp:224:46: error: 'const class YAML::Node' has no member named 'Mark'
         target.setLoc(node.Mark().line, node.Mark().column);
                                              ^~~~
src/base/AnyMap.cpp:239:44: error: 'const class YAML::Node' has no member named 'Mark'
             const auto& loc = child.second.Mark();
                                            ^~~~
src/base/AnyMap.cpp: In static member function 'static bool YAML::convert<Cantera::AnyValue>::decode(const YAML::Node&, Cantera::AnyValue&)':
src/base/AnyMap.cpp:414:28: error: 'const class YAML::Node' has no member named 'Mark'
         target.setLoc(node.Mark().line, node.Mark().column);
                            ^~~~
src/base/AnyMap.cpp:414:46: error: 'const class YAML::Node' has no member named 'Mark'
         target.setLoc(node.Mark().line, node.Mark().column);
                                              ^~~~
scons: *** [build/src/base/AnyMap.os] Error 1
scons: building terminated because of errors.
```

It appears that the `Mark` member was released in 0.5.3: https://github.com/jbeder/yaml-cpp/commit/ec8aa4fa625257b0db7254460b91b517e2b16a7e

**System information**

- Cantera version: `main`
- OS: Ubuntu 18.04
- Python/MATLAB/other software versions: System installs of everything, but Python 3 running SCons

--- a/SConstruct
+++ b/SConstruct
@@ -1080,13 +1080,14 @@ except ValueError:
 
 # Check for yaml-cpp library and checkout submodule if needed
 if env['system_yamlcpp'] in ('y', 'default'):
-    if conf.CheckCXXHeader('yaml-cpp/yaml.h', '""'):
+    # We need the Mark() function, which was added in version 0.5.3
+    if conf.CheckStatement('YAML::Node().Mark()', '#include "yaml-cpp/yaml.h"'):
         env['system_yamlcpp'] = True
         print("""INFO: Using system installation of yaml-cpp library.""")
 
     elif env['system_yamlcpp'] == 'y':
-        config_error('Expected system installation of yaml-cpp library, but it '
-            'could not be found.')
+        config_error("Expected system installation of yaml-cpp library, but it "
+            "could not be found or it is too old (0.6 or newer is required).")
 
 if env['system_yamlcpp'] in ('n', 'default'):
     env['system_yamlcpp'] = False

